src/index.ts

```diff
@@ -45,37 +77,28 @@ function main(state: State, env: DotEnv) {
 			}
 		}
 
-		if (message.content.startsWith(config.commandPrefix)) {
-			// Handle commands.
-			// TODO: https://github.com/RSA-Bots/PandaProtector/issues/3
-			const content = message.content.slice(config.commandPrefix.length);
-			const matches = /^(\w+)\s*(.*)/su.exec(content);
-			const commandName = matches?.[1] ?? "";
-			const argumentContent = matches?.[2] ?? "";
-
-			if (commandName.length > 0) {
-				const command = getCommand(commandName);
-
-				if (command && command.hasPermission(state, message)) {
-					const args = command.parseArguments(argumentContent);
-					const required = command.options.reduce((acc, option) => acc + (option.optional ? 0 : 1), 0);
-
-					if (args.length >= required) {
-						command.handler(state, message, ...command.parseArguments(argumentContent));
-					} else {
-						ephemeral(state, message.reply(`Missing arguments for **${commandName}**.`)).catch(
-							console.error
-						);
-					}
+		//Handle owner only async commands
+		(async () => {
+			if (!client.application?.owner) await client.application?.fetch();
+
+			if (message.author.id === client.application?.owner?.id) {
+				if (message.content.toLowerCase() === "!deploy") {
+					deploySlashCommands(client, config, message);
+				} else if (message.content.toLowerCase() === "!unload") {
+					await client.guilds.cache.get(config.guildId)?.commands.set([]);
+					void message.reply("Successfully unloaded slash-commands.");
+				} else if (message.content.toLowerCase() === "!unload-global") {
+					await client.application?.commands.set([]);
+					void message.reply(
+						"Global slash-commands successfully unloaded. Please give approx 1 hour for changes to take effect."
+					);
```

Comment:
This feels a bit hacky to keep it all hardcoded in index.ts but perhaps we should keep it until we learn if slash-commands are stable.

Reply:
We can implement them as slash-commands, and just prevent them from being unloaded. Otherwise, hardcoding is the only way I can think of.

Also, are we changing to devRoleId instead of owner only?

Reply:
Yeah I've already committed that change